PhysicsTools/PatAlgos/python/jetCollectionTools.py

```diff
@@ -0,0 +1,621 @@
+    #========================================================
+    if not inputCollection:
+      print("custom_jme_cff::GenJetAdder::addGenJetCollection: inputCollection in NanoAOD not specified. Building genjet collection now")
+      #
+      # Setup Gen Particles
+      #
+      packedGenPartNoNu = "packedGenParticlesForJetsNoNu"
+      if packedGenPartNoNu not in self.prerequisites:
+        setattr(proc, packedGenPartNoNu, cms.EDFilter("CandPtrSelector",
+            src = cms.InputTag("packedGenParticles"),
+            cut = cms.string("abs(pdgId) != 12 && abs(pdgId) != 14 && abs(pdgId) != 16"),
+          )
+        )
+        self.prerequisites.append(packedGenPartNoNu)    
+      #
+      # Create the GenJet collection
+      #
+      genJetsCollection = "{}{}{}".format(genJetInfo.jetAlgo.upper(), genJetInfo.jetSize, 'GenJetsNoNu')
+      setattr(proc, genJetsCollection,
+        cms.EDProducer("FastjetJetProducer",
+          GenJetParameters.clone(
+            src          = packedGenPartNoNu,
+            doAreaFastjet = cms.bool(True),
+          ),
+          AnomalousCellParameters,
+          jetAlgorithm = cms.string(supportedJetAlgos[genJetInfo.jetAlgo]),
+          rParam       = cms.double(genJetInfo.jetSizeNr),
+        )
+      )
+      self.prerequisites.append(genJetsCollection)
+
+    ############################
+    #
+    # Tables
+    #
+    ############################
+    genTable = "{}Table".format(tagName)
+    if genTable in self.main:
+      raise ValueError("Step '%s' already implemented" % genTable)
+
+    setattr(proc, genTable, cms.EDProducer("SimpleCandidateFlatTableProducer",
+        src       = cms.InputTag(genJetsCollection),
+        cut       = cms.string(""),
+        name      = cms.string(name),
+        doc       = cms.string('{} (generator level)'.format(doc)),
+        singleton = cms.bool(False),
+        extension = cms.bool(False),
+        variables = cms.PSet(P4Vars,
+        area      = jetTable.variables.area,
+        ),
+      )
+    )    
+    currentTasks.append(genTable)
+    #
+    # GenJet Flavour Labelling
+    #
+    genFlavour = "{}Flavour".format(tagName)
+    if genFlavour in self.main:
+      raise ValueError("Step '%s' already implemented" % genFlavour)
+    setattr(proc, genFlavour, cms.EDProducer("JetFlavourClustering",
+        jets                     = cms.InputTag(genJetsCollection),
+        bHadrons                 = cms.InputTag("patJetPartons", "bHadrons"),
+        cHadrons                 = cms.InputTag("patJetPartons", "cHadrons"),
+        partons                  = cms.InputTag("patJetPartons", "physicsPartons"),
+        leptons                  = cms.InputTag("patJetPartons", "leptons"),
+        jetAlgorithm             = cms.string(supportedJetAlgos[genJetInfo.jetAlgo]),
+        rParam                   = cms.double(genJetInfo.jetSizeNr),
+        ghostRescaling           = cms.double(1e-18),
+        hadronFlavourHasPriority = cms.bool(False),
```

Comment:
this looks like can be cloned with only [3] relevant parameters updated from mcMatchLayer0.jetFlavourId_cff.patJetFlavourAssociation